stm/vernac_classifier.ml

```diff
@@ -120,19 +119,29 @@ let rec classify_vernac e =
           CList.map_filter (function (Some ((_,i),pl), _) -> Some i | _ -> None) l in
         VtStartProof (default_proof_mode (),GuaranteesOpacity,ids), VtLater
     | VernacGoal _ -> VtStartProof (default_proof_mode (),GuaranteesOpacity,[]), VtLater
-    | VernacFixpoint (_,l) ->
+    | VernacFixpoint (discharge,l) ->
+       let guarantee =
+         match discharge with
+         | Decl_kinds.NoDischarge -> GuaranteesOpacity
+         | Decl_kinds.Discharge -> Doesn'tGuaranteeOpacity
+       in
```

Comment:
This is a fix compared to previous version?

Reply:
Yes, see commit message.